src/auth/strategies/jwt.strategy.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { PassportStrategy } from '@nestjs/passport'
+import { ExtractJwt, Strategy } from 'passport-jwt'
+
+import { throwMissingEnvVar } from '../../common/utils/env.utils'
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(configService: ConfigService) {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      ignoreExpiration: false,
+      secretOrKey:
+        configService.get<string>('JWT_SECRET') ||
+        throwMissingEnvVar('JWT_SECRET'),
+    })
+  }
+
+  async validate({ sub, username }) {
```

Comment:
What's the purpose of this method?

Reply:
NestJS docs: 
_"As a result of all this, our response to the validate() callback is trivial: we simply return an object containing the userId and username properties. Recall again that Passport will build a user object based on the return value of our validate() method, and attach it as a property on the Request object."_

Or, maybe, this question is a bit deeper than just refreshing for me what this method does?

Reply:
I'm not so deep in `@nestjs/passport` library, so the purpose of this method (which is trivial mapper here) wasn't clear to me.
But now I see, thanks.

Considering that `validate` method should return the user object that will be assigned to the `Request`, I see wrong return value here (or wrong types): you've typed `request.user` as
```TS
{
  _id: ObjectId;
  username: string;
  password: string;
}
```
but instead returning:
```TS
{
  userId: string;
  username: string;
}
```

Reply:
Yeah, got it. Also, diving deeper into it, I noticed such behavior: doesn't matter what exactly I put into return value from the `validate()` method, the only improtant moment - it should be a truthy value. I can even return this:
```
async validate() {
    return true
}
```
And checking it into a random request like
```
@Get()
  async findAllUsers(@Request() req): Promise<User[]> {
    console.log(req.user) // true
    return await this.userService.findAllUsers()
  }
```
It returns (as expected) as a `user` field on the every request protected by the jwt strategy. For example, if I return any falsy value here - I'll got an unauthorized error. No `Request` decorator used in my endpoints except the `login` route (no jwt used here, it's a local strategy), so it was an easy way to miss it. So, agreed again with your previous observations that it's a not the best naming of the `validate` method since it already caused so many confusions.

Reply:
Sooo I'm very confused now. Does it mean that we can return plain `true`? How should we decide when return `true` vs `false`?
I f we don't return user object, then how it gets assigned to the request?

Reply:
Discussed at the online code review call ✅